packages/ckeditor5-engine/src/model/schema.ts

```diff
@@ -1718,6 +1853,25 @@ export class SchemaContext implements Iterable<SchemaContextItem> {
 		return ctx;
 	}
 
+	/**
+	 * Returns a new schema context that is based on this context but has the last item removed.
+	 *
+	 * ```ts
+	 * const ctxParagraph = new SchemaContext( [ '$root', 'blockQuote', 'paragraph' ] );
+	 * const ctxBlockQuote = ctxParagraph.pop(); // Items in `ctxBlockQuote` are: `$root` an `blockQuote`.
+	 * const ctxRoot = ctxBlockQuote.pop(); // Items in `ctxRoot` are: `$root`.
+	 * ```
+	 *
+	 * @returns A new reduced schema context instance.
+	 */
+	public pop(): SchemaContext {
```

Comment:
I used `pop()` to have analogy with `push()` method that is available on the `SchemaContext` already. I don't like it either. I don't like `trim()` and `trimEnd()` as it returns "empty" values and "end" is uncountable (how much end, how many items?). But I like `trimLast()` it specifically tells you that it will trim the last element.